src/BlockEntities/ChestEntity.cpp

```diff
@@ -38,7 +40,11 @@ void cChestEntity::SendTo(cClientHandle & a_Client)
 	// The chest entity doesn't need anything sent to the client when it's created / gets in the viewdistance
 	// All the actual handling is in the cWindow UI code that gets called when the chest is rclked
 
-	UNUSED(a_Client);
+	// Temporary fix for invisible chests (#3479, #3403)
+	if (a_Client.GetProtocolVersion() >= cProtocolRecognizer::PROTO_VERSION_1_9_0) 
+	{
+		a_Client.SendUpdateBlockEntity(*this);
+	}
```

Comment:
I meant that it would make more sense to try this:
```cpp
// Send a dummy "number of players with chest open" packet to make the chest visible:
a_Client.SendBlockAction(m_PosX, m_PosY, m_PosZ, 1, 0, m_BlockType);
```
I did try and the chests are no longer invisible, it works perfectly and doesn't use any undocumented packet.